apps/voting/test/voting.js

```diff
@@ -174,189 +139,234 @@ contract('Voting App', accounts => {
-                assert.equal(await executionTarget.counter(), 2, 'should have executed result')
-            })
+                it('throws when non-holder votes', async () => {
+                    return assertRevert(async () => {
+                        await voting.vote(voteId, true, true, { from: nonHolder })
+                    })
+                })
 
-            it('cannot execute vote if not enough quorum met', async () => {
-                await voting.vote(voteId, true, true, { from: holder19 })
-                await timeTravel(votingTime + 1)
-                return assertRevert(async () => {
-                    await voting.executeVote(voteId)
+                it('throws when voting after voting closes', async () => {
+                    await timeTravel(votingTime + 1)
+                    return assertRevert(async () => {
+                        await voting.vote(voteId, true, true, { from: holder29 })
+                    })
                 })
-            })
 
-            it('cannot execute vote if not support met', async () => {
-                await voting.vote(voteId, false, true, { from: holder31 })
-                await voting.vote(voteId, false, true, { from: holder19 })
-                await timeTravel(votingTime + 1)
-                return assertRevert(async () => {
+                it('can execute if vote is approved with support and quorum', async () => {
+                    await voting.vote(voteId, true, true, { from: holder29 })
+                    await voting.vote(voteId, false, true, { from: holder20 })
+                    await timeTravel(votingTime + 1)
                     await voting.executeVote(voteId)
+                    assert.equal(await executionTarget.counter(), 2, 'should have executed result')
                 })
-            })
 
-            it('vote can be executed automatically if decided', async () => {
-                await voting.vote(voteId, true, true, { from: holder50 }) // causes execution
-                assert.equal(await executionTarget.counter(), 2, 'should have executed result')
-            })
+                it('cannot execute vote if not enough quorum met', async () => {
+                    await voting.vote(voteId, true, true, { from: holder20 })
+                    await timeTravel(votingTime + 1)
+                    return assertRevert(async () => {
+                        await voting.executeVote(voteId)
+                    })
+                })
 
-            it('vote can be not executed automatically if decided', async () => {
-                await voting.vote(voteId, true, false, { from: holder50 }) // doesnt cause execution
-                await voting.executeVote(voteId)
-                assert.equal(await executionTarget.counter(), 2, 'should have executed result')
-            })
+                it('cannot execute vote if not support met', async () => {
+                    await voting.vote(voteId, false, true, { from: holder29 })
```

Comment:
Would be interesting to test when a vote gets 50% support but doesn't pass because of the strict check.